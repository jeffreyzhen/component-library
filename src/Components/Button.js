import React from 'react';

export const Button = () => <button type="submit">Submit</button>;

export default Button;
