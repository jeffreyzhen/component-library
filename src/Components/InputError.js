import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

const StyledError = styled.div`
  color: red;
`;

export const InputError = ({ children }) => (
  <StyledError data-testid="input-error">
    <p>{children}</p>
  </StyledError>
);

InputError.propTypes = {
  children: PropTypes.string.isRequired,
};

export default InputError;
