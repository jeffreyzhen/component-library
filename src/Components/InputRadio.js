import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

const StyledInput = styled.input``;

const StyledLabel = styled.label``;

export const InputRadio = ({
  id,
  name,
  value,
  children,
  onChange,
  activeSelection,
  ...props
}) => {
  return (
    <StyledLabel htmlFor={id}>
      <StyledInput
        type="radio"
        id={id}
        name={name}
        value={value}
        onChange={onChange}
        checked={activeSelection === value}
        {...props}
      />
      {children}
    </StyledLabel>
  );
};

InputRadio.defaultProps = {
  activeSelection: '',
  onChange: () => {},
};

InputRadio.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  onChange: PropTypes.func,
  activeSelection: PropTypes.string,
};

export default InputRadio;
