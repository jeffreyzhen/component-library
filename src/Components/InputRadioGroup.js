import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

const StyledDiv = styled.div`
  display: flex;
`;

export const InputRadioGroup = ({ children, ...props }) => {
  const [activeSelection, setActiveSelection] = useState('');

  const handleChange = event => {
    setActiveSelection(event.target.value);
  };

  const childrenHOC = React.Children.map(children, child => {
    return React.cloneElement(child, {
      onChange: handleChange,
      activeSelection,
    });
  });

  return <StyledDiv {...props}>{childrenHOC}</StyledDiv>;
};

InputRadioGroup.propTypes = {
  children: PropTypes.node.isRequired,
};

export default InputRadioGroup;
