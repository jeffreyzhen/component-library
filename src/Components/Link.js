import React from 'react';

const Link = () => <a href="asdf">Link</a>;

export default Link;
