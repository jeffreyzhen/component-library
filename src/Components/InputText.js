import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

import { InputError } from './InputError';

const StyledDiv = styled.div`
  display: flex;
  flex-direction: column;
`;

const StyledInputText = styled.input``;

const StyledLabel = styled.label``;

export const InputText = ({ id, validate, label, required, ...props }) => {
  const [inputIsPristine, setInputIsPristine] = useState(true);
  const [value, setValue] = useState('');
  const [inputError, setInputError] = useState('');
  const inputHasError = !inputIsPristine && inputError;

  const handleErrorMessage = userInput => {
    if (required && validate === null) {
      if (userInput === '') {
        setInputError(required);
      } else {
        setInputError('');
      }
    }

    if (validate && required === null) {
      if (userInput === '' || validate.regex.test(userInput)) {
        setInputError('');
      } else {
        setInputError(validate.message);
      }
    }

    if (validate && required) {
      if (userInput === '') {
        setInputError(required);
      } else if (validate.regex.test(userInput)) {
        setInputError('');
      } else {
        setInputError(validate.message);
      }
    }
  };

  const handleChange = event => {
    setInputIsPristine(false);
    setValue(event.target.value);
    handleErrorMessage(event.target.value);
  };

  return (
    <StyledDiv>
      <StyledLabel htmlFor={id}>{label}</StyledLabel>
      <StyledInputText
        id={id}
        type="text"
        onChange={handleChange}
        value={value}
        {...props}
      />
      {inputHasError && <InputError>{inputError}</InputError>}
    </StyledDiv>
  );
};

InputText.defaultProps = {
  validate: null,
  required: null,
};

InputText.propTypes = {
  validate: PropTypes.shape({
    regex: PropTypes.instanceOf(RegExp),
    message: PropTypes.string,
  }),
  label: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  required: PropTypes.string,
};

export default InputText;
