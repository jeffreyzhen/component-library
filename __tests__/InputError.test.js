import React from 'react';
import { render } from 'react-testing-library';
import { matchers } from 'jest-emotion';

import { InputError } from '../src/Components/InputError';

expect.extend(matchers);

test('renders correctly', () => {
  const errorMessage = 'This is an error';
  const { getByTestId, container } = render(
    <InputError>{errorMessage}</InputError>
  );

  const error = getByTestId('input-error');
  expect(error).toHaveTextContent(errorMessage);
  expect(error).toHaveStyleRule('color', 'red');
  expect(container.firstChild).toMatchSnapshot();
});
