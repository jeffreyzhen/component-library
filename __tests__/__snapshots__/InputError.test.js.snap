// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.emotion-0 {
  color: red;
}

<div
  class="emotion-0 emotion-1"
  data-testid="input-error"
>
  <p>
    This is an error
  </p>
</div>
`;
